cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -316,20 +316,23 @@ public KafkaAssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pfa,
                 .compose(state -> state.kafkaExporterService())
                 .compose(state -> state.kafkaExporterReady())
 
-                .compose(state -> {
-                    chainPromise.complete();
-                    return chainPromise.future();
+                .compose(state -> Future.succeededFuture())
+                .setHandler(result -> {
+                    if (result.succeeded()) {
+                        chainPromise.complete();
+                    } else {
+                        chainPromise.fail(result.cause());
+                    }
```

Comment:
Why can't this be replaced with `setHandler(chainPromise)`

Reply:
This is because the `compose` works with `Future<ReconciliationState>` but the `chainPromise` is `Future<Void>`. So you cannot do just `setHandler(chainPromise)`.

Reply:
`map((Void) null)` is what you're looking for.

Reply:
I think this has much better readability.

Reply:
Well sure, if you want to read 6 lines rather than one.

Reply:
Well, I want to understand what the code does. One way or another, I'm off now. I'm not going to change it one way or another until Friday.

Reply:
Are you happy for me to do it for you?